src/spark_rapids_ml/core.py

```diff
@@ -438,16 +457,23 @@ def _train_udf(pdf_iter: Iterator[pd.DataFrame]) -> pd.DataFrame:
             if enable_nccl:
                 context.barrier()
 
-            if context.partitionId() == 0:
+            if return_model == True:
+                if context.partitionId() == 0:
+                    yield pd.DataFrame(data=result)
+            else:
```

Comment:
looks like this doesn't require context.barrier in line  439?

Reply:
revised.